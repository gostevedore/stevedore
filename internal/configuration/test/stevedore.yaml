tree_path: stevedore.yaml
builder_path: stevedore.yaml
num_workers: 10
build_skip_push_image: false
build_on_cascade: true
