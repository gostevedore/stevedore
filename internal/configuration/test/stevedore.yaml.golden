#
# Stevedore manages and governs the Docker's image's building process

#
# Images tree location path
#  default value:
#    images_path: stevedore.yaml
#
# images_path: stevedore.yaml

#
# Builders location path
#  default value:
#    builders_path: stevedore.yaml
#
# builders_path: stevedore.yaml

#
# Log file location path
#  default value: 
#    log_path: /dev/null
log_path: test_stevedore.log

#
# It defines the number of workers to build images which corresponds to the number of images that can be build concurrently
#  default value: 
#    concurrency: 4
concurrency: 8

#
# On build, push images automatically after it finishes
#  default value: 
#    push_images: false
#
# push_images: false

#
# Directory to store docker registry credentials
#  default value: 
#    docker_registry_credentials_dir: ~/.config/stevedore/credentials
docker_registry_credentials_dir: .credentials

#
# Generate extra tags when the main image tags is semver 2.0.0 compliance
#  default value: false
#    semantic_version_tags_enabled: false
semantic_version_tags_enabled: true

# List of templates which define those extra tags to generate when 'semantic_version_tags_enabled' is enabled
# Parser will use the SemVer struct to generate the template, and all SemVer attributes could be used to define each template
# 
#   // SemVer is a sematinc version representation
#   type SemVer struct {
#   	Major      string
#   	Minor      string
#   	Patch      string
#   	PreRelease string
#   	Build      string
#   }
#
#  default value: 
#    semantic_version_tags_templates:
#      - {{ .Major }}.{{ .Minor }}.{{ .Patch }}
semantic_version_tags_templates:
- {{ .Major }}
- {{ .Major }}.{{ .Minor }}

#
# Define builder types
# You could define builders on its own file. Stevedore will look up for builders on the file set at 'builders_path'
# 
# examples:
#   1) Define one builder named 'infrastructure' which use docker as driver and current folder as Docker build context
#
#      builders:
#       infrastructure:
#         driver: docker
#         options:
#           context:
#              path: .

#
# Define images tree
# You could define an images tree on its own file. Stevedore will look up for images tree on the file set at 'images_path'
# 
#  examples:
#    1) Define an images tree that has one image named 'ubuntu' which has two versions, '20.04' and '18.04'.
#       The image '20.04' has two children: 'php-fpm' and 'php-cli', both having the version '7.4' defined. And the image '18.04' has only one child: 'php-fpm', with version '7.3' defined.
#       
#       images:
#         ubuntu:
#           18.04:
#             builder: infrastructure
#             children:
#             - php-fpm:
#               - 7.3
#           20.04:
#             builder: infrastructure
#             children:
#             - php-fpm:
#               - 7.4
#             - php-cli:
#               - 7.4
#         php-fpm:
#           7.3:
#             builder: infrastructure
#             builder:
#               driver: docker
#               context:
#                 path: php-fpm
#               vars:
#                 version: 7.3
#           7.4:
#             builder: infrastructure
#             builder:
#               driver: docker
#               context:
#                 path: php-fpm
#               vars:
#                 version: 7.4
#         php-cli:
#           7.4:
#             builder: infrastructure
#             builder:
#               driver: docker
#               context:
#                 path: php-fpm
#               vars:
#                 version: 7.4

